src/frontend/screens/Sync/ProjectSyncDisplay.tsx

```diff
@@ -0,0 +1,274 @@
+import {ScreenContentWithDock} from '../../sharedComponents/ScreenContentWithDock';
+import {Button} from '../../sharedComponents/Button';
+import {Text} from '../../sharedComponents/Text';
+import {StopIcon, SyncIcon, WifiIcon} from '../../sharedComponents/icons';
+import {useQueryClient} from '@tanstack/react-query';
+import {OBSERVATION_KEY} from '../../hooks/server/observations';
+import {useFocusEffect} from '@react-navigation/native';
+
+const m = defineMessages({
+  deviceName: {
+    id: 'screens.Sync.ProjectSyncDisplay.deviceName',
+    defaultMessage: 'Your device name is {name}',
+  },
+  devicesNearby: {
+    id: 'screens.Sync.ProjectSyncDisplay.devicesNearby',
+    defaultMessage:
+      '{count} {count, plural, one {device} other {devices}} nearby/connected',
+  },
+  buttonTextSync: {
+    id: 'screens.Sync.ProjectSyncDisplay.buttonTextSync',
+    defaultMessage: 'Sync',
+  },
+  buttonTextStop: {
+    id: 'screens.Sync.ProjectSyncDisplay.buttonTextStop',
+    defaultMessage: 'Stop',
+  },
+  buttonTextDone: {
+    id: 'screens.Sync.ProjectSyncDisplay.buttonTextDone',
+    defaultMessage: "You're all caught up",
+  },
+  noDevicesSyncing: {
+    id: 'screens.Sync.ProjectSyncDisplay.noDevicesSyncing',
+    defaultMessage: 'No Devices are Syncing',
+  },
+  devicesWaitingToSync: {
+    id: 'screens.Sync.ProjectSyncDisplay.devicesWaitingToSync',
+    defaultMessage:
+      '{count} {count, plural, one {Device} other {Devices}} Waiting to Sync with you',
+  },
+  devicesSyncing: {
+    id: 'screens.Sync.ProjectSyncDisplay.devicesSyncing',
+    defaultMessage:
+      'Syncing with {count} {count, plural, one {Device} other {Devices}}',
+  },
+  upToDate: {
+    id: 'screens.Sync.ProjectSyncDisplay.upToDate',
+    defaultMessage: 'Up to Date!\nNo data to Sync',
+  },
+  syncing: {
+    id: 'screens.Sync.ProjectSyncDisplay.syncing',
+    defaultMessage: 'Syncing…',
+  },
+  syncProgress: {
+    id: 'screens.Sync.ProjectSyncDisplay.syncProgress',
+    defaultMessage: '{value}%',
+  },
+});
+
+export const ProjectSyncDisplay = ({
+  syncState,
+  projectName,
+  deviceName,
+}: {
+  syncState: SyncState;
+  projectName: string;
+  deviceName: string;
+}) => {
+  const {formatMessage: t} = useIntl();
+
+  const project = useProject();
+  const queryClient = useQueryClient();
+
+  const {connectedPeers, data, initial} = syncState;
+  const isSyncDone = !initial.dataToSync && !data.dataToSync;
+
+  // stops sync when user leaves sync screen. The api allows us to continue syncing even if the user is not on the sync screen, but for simplicity we are only allowing sync while on the sync screen. In the future we can easily enable background sync, there are just some UI questions that need to answered before we do that.
+  useFocusEffect(
+    React.useCallback(() => {
+      return () => {
+        project.$sync.stop();
+        queryClient.invalidateQueries({queryKey: [OBSERVATION_KEY]});
+      };
+    }, [project, queryClient]),
+  );
```

Comment:
this seems to be advised against by React Navigation based on what's described here: https://reactnavigation.org/docs/use-focus-effect/#when-to-use-focus-and-blur-events-instead

seems like the preferred approach is using what's described there (i.e. use `blur`), although i think in this specific case, using [`beforeRemove`](https://reactnavigation.org/docs/navigation-events#beforeremove) makes more sense since we want this to happen where you're definitely leaving the screen (iirc, `blur` can be triggered in cases where you don't actually leave the screen)